Micronaut 4: Scheduled job with fixedRate from configuration AND condition does not work

### Expected Behavior

The following job should run and pick up the fixed rate value from the configuration (yaml, properties, env variables ...).

```java
@Singleton
public class SomeJob {

    private boolean paused = false;

    @Scheduled(fixedRate = "${hello.world.rate}", condition = "#{!this.paused}" )
    public void doSomething() {
        System.out.println("The job runs...");
    }

    public boolean isPaused() {
        return paused;
    }
}
```

### Actual Behaviour

The job above cause Micronaut to fail while bootstrapping the application. For some reason the expression `${hello.world.rate}` is not resolved.

```
|  \/  (_) ___ _ __ ___  _ __   __ _ _   _| |_ 
| |\/| | |/ __| '__/ _ \| '_ \ / _` | | | | __|
| |  | | | (__| | | (_) | | | | (_| | |_| | |_ 
|_|  |_|_|\___|_|  \___/|_| |_|\__,_|\__,_|\__|
13:04:43.767 [ForkJoinPool.commonPool-worker-7] ERROR i.m.context.DefaultBeanContext - Error processing bean method Definition: io.wangler.micronaut.jobs.SomeJob.void doSomething() with processor (io.micronaut.scheduling.processor.ScheduledMethodProcessor@25f2dbf): Invalid @Scheduled definition for method: void doSomething() - Reason: Invalid fixed rate definition: ${hello.world.rate}
io.micronaut.scheduling.exceptions.SchedulerConfigurationException: Invalid @Scheduled definition for method: void doSomething() - Reason: Invalid fixed rate definition: ${hello.world.rate}
	at io.micronaut.scheduling.processor.ScheduledMethodProcessor.lambda$process$3(ScheduledMethodProcessor.java:161)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at io.micronaut.scheduling.processor.ScheduledMethodProcessor.process(ScheduledMethodProcessor.java:160)
	at io.micronaut.context.DefaultBeanContext.lambda$initializeContext$31(DefaultBeanContext.java:2032)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
13:04:43.783 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Failed to inject value for parameter [localeResolver] of class: io.micronaut.http.server.binding.LocaleArgumentBinder

Message: No bean of type [io.micronaut.http.server.util.locale.HttpLocaleResolver] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
Path Taken: new DefaultRequestBinderRegistry(ConversionService conversionService,List binders,DefaultBodyAnnotationBinder bodyAnnotationBinder) --> new DefaultRequestBinderRegistry(ConversionService conversionService,[List binders],DefaultBodyAnnotationBinder bodyAnnotationBinder) --> new LocaleArgumentBinder([HttpLocaleResolver localeResolver])
io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [localeResolver] of class: io.micronaut.http.server.binding.LocaleArgumentBinder

Message: No bean of type [io.micronaut.http.server.util.locale.HttpLocaleResolver] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
Path Taken: new DefaultRequestBinderRegistry(ConversionService conversionService,List binders,DefaultBodyAnnotationBinder bodyAnnotationBinder) --> new DefaultRequestBinderRegistry(ConversionService conversionService,[List binders],DefaultBodyAnnotationBinder bodyAnnotationBinder) --> new LocaleArgumentBinder([HttpLocaleResolver localeResolver])
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2182)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConstructorArgument(AbstractInitializableBeanDefinition.java:1328)
	at io.micronaut.http.server.binding.$LocaleArgumentBinder$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2307)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2277)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2289)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3056)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:81)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2958)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2919)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2893)
	at io.micronaut.context.DefaultBeanContext.addCandidateToList(DefaultBeanContext.java:3521)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistrations(DefaultBeanContext.java:3473)
	at io.micronaut.context.DefaultBeanContext.getBeanRegistrations(DefaultBeanContext.java:3443)
	at io.micronaut.context.DefaultBeanContext.getBeansOfType(DefaultBeanContext.java:1436)
	at io.micronaut.context.AbstractBeanResolutionContext.getBeansOfType(AbstractBeanResolutionContext.java:95)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBeansOfType(AbstractInitializableBeanDefinition.java:2254)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeansOfTypeForConstructorArgument(AbstractInitializableBeanDefinition.java:1476)
	at io.micronaut.http.bind.$DefaultRequestBinderRegistry$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2307)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2277)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2289)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3056)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:81)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2958)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2919)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2730)
	at io.micronaut.context.DefaultBeanContext.findBean(DefaultBeanContext.java:1756)
	at io.micronaut.context.DefaultBeanContext.findBean(DefaultBeanContext.java:1731)
	at io.micronaut.context.DefaultBeanContext.findBean(DefaultBeanContext.java:856)
	at io.micronaut.context.BeanLocator.findBean(BeanLocator.java:346)
	at io.micronaut.http.filter.BaseFilterProcessor.<init>(BaseFilterProcessor.java:81)
	at io.micronaut.web.router.ServerFilterRouteBuilder$1.<init>(ServerFilterRouteBuilder.java:64)
	at io.micronaut.web.router.ServerFilterRouteBuilder.<init>(ServerFilterRouteBuilder.java:64)
	at io.micronaut.web.router.$ServerFilterRouteBuilder$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2307)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2277)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2289)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3056)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:81)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2958)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2919)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2893)
	at io.micronaut.context.DefaultBeanContext.addCandidateToList(DefaultBeanContext.java:3521)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistrations(DefaultBeanContext.java:3473)
	at io.micronaut.context.DefaultBeanContext.getBeanRegistrations(DefaultBeanContext.java:3443)
	at io.micronaut.context.DefaultBeanContext.getBeansOfType(DefaultBeanContext.java:1436)
	at io.micronaut.context.AbstractBeanResolutionContext.getBeansOfType(AbstractBeanResolutionContext.java:95)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBeansOfType(AbstractInitializableBeanDefinition.java:2254)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeansOfTypeForConstructorArgument(AbstractInitializableBeanDefinition.java:1476)
	at io.micronaut.web.router.$DefaultRouter$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2307)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2277)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2289)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3056)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:81)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2958)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2919)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2730)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1693)
	at io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:89)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2166)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConstructorArgument(AbstractInitializableBeanDefinition.java:1328)
	at io.micronaut.http.server.$RouteExecutor$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2307)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2277)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2289)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3056)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:81)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2958)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2919)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2730)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1693)
	at io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:89)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2166)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConstructorArgument(AbstractInitializableBeanDefinition.java:1328)
	at io.micronaut.http.server.netty.$DefaultNettyEmbeddedServerFactory$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2307)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2277)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2289)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3056)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:81)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2958)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2919)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2730)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1693)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1675)
	at io.micronaut.http.server.netty.$DefaultNettyEmbeddedServerFactory$BuildDefaultServer0$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2307)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2277)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2289)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3056)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:81)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2958)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2919)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2730)
	at io.micronaut.context.DefaultBeanContext.findBean(DefaultBeanContext.java:1756)
	at io.micronaut.context.DefaultBeanContext.findBean(DefaultBeanContext.java:1731)
	at io.micronaut.context.DefaultBeanContext.findBean(DefaultBeanContext.java:856)
	at io.micronaut.context.BeanLocator.findBean(BeanLocator.java:346)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:75)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:322)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:308)
	at io.wangler.micronaut.Application.main(Application.java:8)
Caused by: io.micronaut.context.exceptions.NoSuchBeanException: No bean of type [io.micronaut.http.server.util.locale.HttpLocaleResolver] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
	at io.micronaut.context.DefaultBeanContext.newNoSuchBeanException(DefaultBeanContext.java:2773)
	at io.micronaut.context.DefaultApplicationContext.newNoSuchBeanException(DefaultApplicationContext.java:292)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2735)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1693)
	at io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:89)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2166)
	... 103 common frames omitted

```

### Steps To Reproduce

1. Clone repo https://github.com/saw303/m4configuration
2. Checkout branch `micronaut400_scheduled`
3. Run the application

### Environment Information

_No response_

### Example Application

https://github.com/saw303/m4configuration

### Version

4.0.1

--- a/core/src/main/java/io/micronaut/core/annotation/AnnotationValue.java
+++ b/core/src/main/java/io/micronaut/core/annotation/AnnotationValue.java
@@ -183,6 +183,7 @@ public class AnnotationValue<A extends Annotation> implements AnnotationValueRes
 
     /**
      * @return The value mapper.
+     * @since 4.0.2
      */
     protected @Nullable Function<Object, Object> getValueMapper() {
         return valueMapper;

--- a/core/src/main/java/io/micronaut/core/annotation/AnnotationValue.java
+++ b/core/src/main/java/io/micronaut/core/annotation/AnnotationValue.java
@@ -182,6 +182,13 @@ public class AnnotationValue<A extends Annotation> implements AnnotationValueRes
     }
 
     /**
+     * @return The value mapper.
+     */
+    protected @Nullable Function<Object, Object> getValueMapper() {
+        return valueMapper;
+    }
+
+    /**
      * Creates a builder with the initial value of this annotation.
      *
      * @return The builder with this annotation value

--- a/inject-java/src/test/groovy/io/micronaut/expressions/EnvironmentAccessExpressionsSpec.groovy
+++ b/inject-java/src/test/groovy/io/micronaut/expressions/EnvironmentAccessExpressionsSpec.groovy
@@ -2,8 +2,51 @@ package io.micronaut.expressions
 
 import io.micronaut.annotation.processing.test.AbstractEvaluatedExpressionsSpec
 import io.micronaut.context.env.PropertySource
+import io.micronaut.inject.ExecutableMethod
+import io.micronaut.scheduling.annotation.Scheduled
+import spock.lang.Issue
 
 class EnvironmentAccessExpressionsSpec extends AbstractEvaluatedExpressionsSpec {
+
+    @Issue('https://github.com/micronaut-projects/micronaut-core/issues/9622#issuecomment-1654002320')
+    void "test environment access expression with placeholder expression"() {
+        given:
+        def ctx = buildContext("""
+            package test;
+
+            import io.micronaut.scheduling.annotation.Scheduled;
+
+            @jakarta.inject.Singleton
+            class MyJob {
+
+                private boolean paused = false;
+
+                @Scheduled(fixedRate = "\${hello.world.rate}", condition = "#{!this.paused}" )
+                public void doSomething() {
+                    System.out.println("The job runs...");
+                }
+
+                public boolean isPaused() {
+                    return paused;
+                }
+
+            }
+        """)
+
+        def type = ctx.classLoader.loadClass('test.MyJob')
+
+        ctx.environment.addPropertySource(PropertySource.of("test",
+                ['hello.world.rate': '100s']))
+
+        def bean = ctx.getBeanDefinition(type)
+        def method = bean.getRequiredMethod("doSomething")
+
+        expect:
+        method.getAnnotationValuesByType(Scheduled).get(0).stringValue("fixedRate").get() == '100s'
+
+        cleanup:
+        ctx.close()
+    }
 
     void "test environment access"() {
         given:

--- a/inject/src/main/java/io/micronaut/inject/annotation/EnvironmentAnnotationValue.java
+++ b/inject/src/main/java/io/micronaut/inject/annotation/EnvironmentAnnotationValue.java
@@ -25,6 +25,7 @@ import java.lang.annotation.Annotation;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
@@ -85,6 +86,11 @@ class EnvironmentAnnotationValue<A extends Annotation> extends AnnotationValue<A
             return o;
         } : null);
         this.environment = environment;
+    }
+
+    @Override
+    protected Function<Object, Object> getValueMapper() {
+        return super.getValueMapper();
     }
 
     @Override

--- a/inject/src/main/java/io/micronaut/inject/annotation/EvaluatedAnnotationValue.java
+++ b/inject/src/main/java/io/micronaut/inject/annotation/EvaluatedAnnotationValue.java
@@ -22,6 +22,7 @@ import io.micronaut.core.annotation.Nullable;
 import io.micronaut.core.expressions.EvaluatedExpression;
 
 import java.lang.annotation.Annotation;
+import java.util.function.Function;
 
 /**
  * An EvaluatedAnnotationValue is a {@link AnnotationValue} that contains one or more expressions.
@@ -42,6 +43,11 @@ public final class EvaluatedAnnotationValue<A extends Annotation> extends Annota
             value -> {
                 if (value instanceof EvaluatedExpression expression) {
                     return expression.evaluate(evaluationContext);
+                } else if (annotationValue instanceof EnvironmentAnnotationValue<A> eav) {
+                    Function<Object, Object> valueMapper = eav.getValueMapper();
+                    if (valueMapper != null) {
+                        return valueMapper.apply(value);
+                    }
                 }
                 return value;
             }

